File: CPP/CWE-477/25/bad.cpp
/*
 * Copyright © 2018, Jon Hood www.hoodsecurity.com
 * License: ISC
 */
#include <cstdlib>
#include <iostream>
#include <vector>
#include <string>
#include <algorithm>
#include <iterator>

using namespace std;

int main(int argc, char *argv[])
{
	vector<string> v = {"one", "two", "three"};
	transform(v.begin(), v.end(), ostream_iterator<size_t>(cout, " "), std::mem_fun_ref(&string::size)); //mem_fun_ref is deprecated; prints size of each word in v
	return EXIT_SUCCESS;
}

File: C#/CWE-476/1/bad.cs
/*
 * Copyright © 2018, Jon Hood www.hoodsecurity.com
 * License: ISC
 */

using System;

namespace BadWeakness
{
	class Bad
	{
		static void Main()
		{
			unsafe
			{
				bool* p = null;
				System.Console.WriteLine(*p);
			}
		}
	}
}
